astronet/astro_cnn_model/input_ds.py

```diff
@@ -93,6 +95,116 @@ def parse_example(serialized_example):
     ds = tf.data.Dataset.from_tensor_slices(filenames)
     ds = ds.flat_map(tf.data.TFRecordDataset)
     ds = ds.map(parse_example)
+
+###### rough code for checking whether the candidate has been assigned disps E,N,B, or S, and only using these candidates in the data augmentation. There is probably a more concise way to do this ######
+
+
+#     values = np.array([x[1].numpy() for x in ds])
+#    truth_array = []
+#    for array in values:
+#        z = array == [1., 1., 1., 1., 1.]
+#        if z[0] or z[1] or z[3] or z[4]:
+#            truth_array.append(True)
+#        else:
+#            truth_array.append(False)       
+#    truth_data = tf.data.Dataset.from_tensor_slices(truth_array)
+#    ds_new = tf.data.Dataset.zip((ds,truth_data))
+#    new_len = [x for x in ds_new]
+#    print("number trues =",np.array(truth_array).sum())
+#    ds_false = ds_new.filter(lambda image,y:y==False)
+#    false_list = [x for x in ds_false]
+#    randoms = []
+#    for i in range(len(false_list)):
+#        randoms.append(random.randint(0,1))
+#    rand = tf.data.Dataset.from_tensor_slices(randoms)
+#    false_rand = tf.data.Dataset.zip((ds_false,rand))
+#    false_rand = false_rand.filter(lambda x,z: z==1)
+#    ds_new = ds_new.filter(lambda image,y: y ==True)
+    
+    
+    
+    globals  = ['global_mask', 'global_std', 'global_transit_mask', 'global_view', 'global_view_0.3', 'global_view_5.0', 'global_view_double_period', 'global_view_double_period_0.3', 'global_view_double_period_5.0']
```

Comment:
I think we don't want to add noise to `mask`s or `std`

but we do want to reverse everything

Reply:
(also applicable to locals / secondaries)